asf_installer_survey/notebooks/section_6_analysis.py

```diff
@@ -0,0 +1,1670 @@
+           )
+
+# %%
+# Generate figure
+donut(df_604_crosstab_owner_V2,
+           "604_owner_V2",
+           "SQ 43a. & 43b. I/my business offers heat pump repairs, servicing and maintenance:"
+           )
+
+# %%
+# CompanySizeEmployee V2
+# Generate crosstab dataframe
+df_604_crosstab_employee_V2 = crosstable("CompanySizeEmployeeV2",
+                                         data,
+                                        "SQ43",
+                                        ans_43
+                                        )
+
+# Save to .csv and display
+df_604_crosstab_employee_V2.to_csv("../../outputs/section6/csv/604_employee_V2.csv")
+df_604_crosstab_employee_V2
+
+# %%
+# Generate figure
+groupedbar(df_604_crosstab_employee_V2,
+           "604_employee_V2",
+           "SQ 43c. The business I work for offers heat pump repairs, servicing and maintenance:"
+           )
+
+# %%
+# Generate figure
+donut(df_604_crosstab_employee_V2,
+           "604_employee_V2",
+           "SQ 43c. The business I work for offers heat pump repairs, servicing and maintenance:"
+           )
+
+# %% [markdown]
+# ### RQ 6.05 | Why do survey participants who only offer repairs, servicing and maintenance for heat pump systems they’ve installed choose not to do so for all other systems?
+# - SQ: 44a, 44b
+# - Sample
+# - Standard_output
+# <p>
+# N.B. Select all that apply type question
+# <p>
+# Update: Free text responses recoded on 2024-04-17
+
+# %%
+# Print possible answers and check counts
+data[col.q44a[0]].value_counts()
+data[col.q44b[0]].value_counts()
+
+# %%
+def transform_44(x):
+    """
+    Function to be applied to dataframe which separates answers selected
+    by creating an individual tally column for each possible answer. 
+
+    Args:
+        x: pandas dataframe containing responses from analytical sample.
+    """
+    
+    # Tally occurrence of each answer
+    if response in x[col.q44a[0]] or response in x[col.q44b[0]]:
+        return response
+    else:
+        return None
+
+# Specifying desired order for outputs
+ans_44 = ["Relationship with existing customers",
+          "No demand from other customers",
+          "Don’t want to take responsibility for other installers’ heat pump installations",
+          "I only work on equipment I understand or have been trained in",
+          "Only work with certain heat pump models", # re-coded new option
+          "Other",
+          "Don't know",
+          "Total"
+          ]
+
+# Create separate dataframe that stacks all answer columns
+df_44 = pd.DataFrame()
+for n in range(0, len(ans_44) - 1):
+
+    # Create new column for each answer option
+    response = ans_44[n]
+    new_col = "SQ44Ans" + str(n + 1)
+    data[new_col] = data.apply(transform_44, axis=1)
+
+    # Stack columns
+    df = data.filter(["EmploymentType", new_col])
+    df = df.rename(columns={new_col:"SQ44"})
+    df_44 = pd.concat([df_44, df])
```

Comment:
Ideally we want more descriptive dataframe names than 'df' and 'df_44', it will just help others when they come to the code!